Auth/Base.php

```diff
@@ -178,7 +178,7 @@ public function getTokenAuthSecret()
         }
 
         if (!empty($this->password)) {
-            return md5($this->password);
+            return hash('sha256', $this->password);
```

Comment:
Isn't that used for comparing the password hash later. In that case it should use the same algo as used to store the password in the database. See `Piwik\Auth\Password::hash()`

Reply:
@sgiehl in this case older passwords will need to be updated too right ?

Reply:
I'm not very deep into the LDAP plugin. Is the password stored in the database actually used for the login? If so it might need to be updated.

Reply:
@sgiehl cant seem to find the usage of `getTokenAuthSecret`, I tried adding a debugger on `plugins/Login/Auth.php` Line 61 and I checked with `md5` as well as `sha` it seems to work with both when I try to login

Reply:
@sgiehl After working on LDAP's other issues, we need to update the password on plugin update, for that we need to run the sync command and set this `Config::getInstance()->LoginLdap['synchronize_users_after_login'] = 1;` before calling that function, which will update all the passwords provided we merge this [PR](https://github.com/matomo-org/plugin-LoginLdap/pull/294)